rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -184,7 +184,7 @@ set_property(
 
 macro(set_properties _build_type)
   set_target_properties(${_target_name} PROPERTIES
-    COMPILE_FLAGS "${_extension_compile_flags}"
+    COMPILE_FLAGS ${_extension_compile_flags}
```

Comment:
Can't use `COMPILE_OPTIONS` here because the evaluation of `_extension_compile_flags` results in a string.
We could work around it by checking the compiler within this macro and make a different call to "set_target_properties" with COMPILE_DEFINITION only if the compiler supports the flags we want